java/ql/lib/semmle/code/java/security/Encryption.qll

```diff
@@ -168,7 +168,7 @@ string getInsecureAlgorithmRegex() {
 string getASecureAlgorithmName() {
   result =
     [
-      "RSA", "SHA256", "SHA512", "CCM", "GCM", "AES([^a-zA-Z](?!ECB|CBC/PKCS[57]Padding)).*",
+      "RSA", "SHA256", "SHA512", "CCM", "GCM", "AES(?![^a-zA-Z](ECB|CBC/PKCS[57]Padding))",
```

Comment:
`.*` does not seem to be necessary here because the `algorithmRegex` predicate will add that.

Also, is there a reason why these queries use `rank[...]` and `max` instead of `concat` to concat the algorithm names? Should I replace that?

Reply:
No good reason AFAICT, it's not important though